File: training-data/objectivec/GcdLcm.m
#import <Foundation/Foundation.h>

NSInteger myGcd(NSInteger a, NSInteger b) {
    while (b != 0) {
        NSInteger t = b;
        b = a % b;
        a = t;
    }
    return labs(a);
}

NSInteger myLcm(NSInteger a, NSInteger b) {
    return labs(a * b) / myGcd(a, b);
}

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        NSLog(@"%ld", (long)myGcd(48, 18));
        NSLog(@"%ld", (long)myLcm(4, 6));
    }
    return 0;
}

File: training-data/prolog/gcd_lcm.pl
my_gcd(A, 0, A) :- !.
my_gcd(A, B, G) :-
    B > 0,
    R is A mod B,
    my_gcd(B, R, G).

my_lcm(A, B, L) :-
    my_gcd(A, B, G),
    L is abs(A * B) // G.

:- my_gcd(48, 18, G), writeln(G).
:- my_lcm(4, 6, L), writeln(L).
